clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetsRequestManager.java

```diff
@@ -431,14 +424,14 @@ protected CompletableFuture<Void> resetPositionsIfNeeded() {
      * detected, a {@link LogTruncationException} will be saved in memory, to be thrown on the
      * next call to this function.
```

Comment:
I suppose that `cachedUpdatePositionsException` is involved here and it might be worth mentioned that in the comment.